packages/marshal/README.md

```diff
@@ -5,127 +5,156 @@ objects) into a string, and back again.
-arguments are omitted, they default to the identity function.
+an object with `toCapData` and `fromCapData` properties. Each callback defaults
+to the identity function.
 
 ```js
 import '@endo/init';
 import { makeMarshal } from '@endo/marshal';
 
 const m = makeMarshal();
 const o = harden({a: 1});
-const s = m.serialize(o);
-console.log(s); // { body: '{"a":1}', slots: [] }
-const o2 = m.unserialize(s);
-console.log(o2); // { a: 1 }
+const s = m.toCapData(o);
+console.log(s);
+// { body: '{"a":1}', slots: [] }
+const o2 = m.fromCapData(s);
+console.log(o2);
+// { a: 1 }
+console.log(o1 === o2);
+// false
 ```
 
 ## Frozen Objects Only
 
 The entire object graph must be "hardened" (recursively frozen), such as done
-by the `ses` module (installed with `@endo/init`). The serialization
-function will refuse to marshal any graph that contains a non-frozen object.
+by the `harden` function installed when importing `@endo/init`. `toCapData` will
+refuse to marshal any object graph that contains a non-frozen object.
 
 ## Beyond JSON
 
-`marshal` uses a special marker object to represent both Presences and data
-which cannot be expressed directly in JSON. This marker uses a property named
-`@qclass` that identifies the type of the object. For example, a Javascript
-`NaN` is serialized into:
+`marshal` uses special values to represent both Presences and data which cannot
+be expressed directly in JSON. These special values are usually strings with
+reserved prefixes in the preferred "smallcaps" encoding, but in the original
+encoding were objects with a property named `@qclass`. For example:
 
 ```
-m.serialize(NaN);
-// { body: '{"@qclass":"NaN"}', slots: [] }
-```
+import '@endo/init';
+import { makeMarshal } from '@endo/marshal';
 
-(TODO) To tolerate a `@qclass` property appearing in the data being
-serialized, the library uses a structure known as a "Hilbert Hotel", which
-wraps the troublesome object in a new layer of serialization.
+// Smallcaps encoding.
+const m1 = makeMarshal(undefined, undefined, { serializeBodyFormat: 'smallcaps' });
+console.log(m1.toCapData(NaN));
+// { body: '#"#NaN"', slots: [] }
 
+// Original encoding.
+const m2 = makeMarshal();
+console.log(m2.toCapData(NaN));
+// { body: '{"@qclass":"NaN"}', slots: [] }
+```
 
 ## Pass-by-Presence vs Pass-by-Copy
 
 `marshal` makes a distinction between objects that are pass-by-presence, and
 those which are pass-by-copy.
 
-To qualify as pass-by-presence, all enumerable properties of the object (and
-of all objects in the inheritance hierarchy) must be methods, not data.
-Pass-by-presence objects usually have identity (assuming the
-`convertValToSlot` and `convertSlotToVal` callbacks behave well), so passing
-the same object through multiple calls will result in multiple references to
-the same output object.
+To qualify as pass-by-presence, all properties of an object (and of all objects
+in its inheritance hierarchy) must be methods, not data. Pass-by-presence objects
```

Comment:
The rules come from the `pass-style` package now, right? It's OK to reiterate them here, but we should refer to the definitive source.

Reply:
Good point. Fixed.